### ISSUE TITLE
When using Space Exploration's navigation sattelite view, attempting to summon a supply train crashes the map

### ISSUE THREAD

**Issue by AmandaCameron** [leehuk/factorio-tnfp#14]
I suspect it is due to `player.character` being nil while in the navigation mode, as [this line](https://github.com/leehuk/factorio-tnfp/blob/master/tnp/message.lua#L65) is where the crash is reported according to the stack trace, going up one is [this call to tnp_speechbubble](https://github.com/leehuk/factorio-tnfp/blob/master/tnp/supplytrain.lua#L38) -- I'm not familiar enough with factorio's modding environment for what could be done about this, though it's worth noting how the supply train selection works now is also sub-optimal when using a map view, since you can't see what train you're summoning.

**Comment by AmandaCameron** [comment]
For reference, here's the full stack trace:

```
The mod Train Network for Players (0.12.4) caused a non-recoverable error.
Please report this error to the mod author.

Error while running event TrainNetworkForPlayers::tnp-handle-railtool-supply (ID 244)
Need entity target
stack traceback:
	[C]: in function 'create_entity'
	__TrainNetworkForPlayers__/tnp/message.lua:65: in function 'tnp_speechbubble'
	__TrainNetworkForPlayers__/tnp/supplytrain.lua:38: in function 'tnp_supplytrain_select'
	__TrainNetworkForPlayers__/tnp/event.lua:189: in function 'tnp_handle_railtool_supply'
	__TrainNetworkForPlayers__/tnp/event.lua:89: in function <__TrainNetworkForPlayers__/tnp/event.lua:83>
```

**Comment by leehuk** [comment]
That should be fixed in 0.12.5 now -- thanks for the report.

I've changed it so if the player has no character, or the players rendering view is the map it'll send console messages instead of speech bubbles.  A custom dialog would be nicer I think.. but they're a bit more work to make them look half decent
